treeherder/etl/schemas/job.yml

```diff
@@ -0,0 +1,278 @@
+      maxLength: 50
+
+
+  # todo: datetime vs unix timestamp
+  # time data
+  timeScheduled:
+    type: "string"
+    format: "date-time"
+  timeStarted:
+    type: "string"
+    format: "date-time"
+  timeCompleted:
+    type: "string"
+    format: "date-time"
+
+  optionCollection:
+    title: "option collection"
+    description: |
+      Options are a dimension of a platform.  The values here can vary wildly,
+      so most strings are valid for this.  The list of options that are used
+      is maleable going forward.
+
+      Some examples of options that have been used:
+        opt    Optimize Compiler GCC optimize flags
+        debug  Debug flags passed in
+        pgo    Profile Guided Optimization - Like opt, but runs with profiling, then builds again using that profiling
+        asan   Address Sanitizer
+        tsan   Thread Sanitizer Build
+    type: "array"
+    items:
+      type: "string"
+      minLength: 1
+      maxLength: 50
+      pattern: "^[A-Za-z0-9_-]+$"
+
+  who:
+    title: "who"
+    format: "email"
+    type: "string"
+    minLength: 1
+    maxLength: 50
+  reason:
+    description: |
+      Examples include:
+      - scheduled
+      - scheduler
+      - Self-serve: Rebuilt by foo@example.com
+      - Self-serve: Requested by foo@example.com
+      - The Nightly scheduler named 'b2g_mozilla-inbound periodic' triggered this build
+      - unknown
+    type: "string"
+    minLength: 1
+    maxLength: 125
+
+  machine:
+    build:
+      type: "object"
+      properties:
+        platform:
+          type: "string"
+          minLength: 1
+          maxLength: 25
+        osName:
+          type: "string"
+          minLength: 1
+          maxLength: 25
+        architecture:
+          type: "string"
+          minLength: 1
+          maxLength: 25
+      required:
+        - platform
+        - os_name
+        - architecture
+    test:
+      type: "object"
+      properties:
+        machineName:
+          type: "string"
+          pattern: "^[A-Za-z0-9_-]+$"
+          minLength: 1
+          maxLength: 50
+        platform:
+          type: "string"
+          minLength: 1
+          maxLength: 25
+        osName:
+          type: "string"
+          minLength: 1
+          maxLength: 25
+        architecture:
+          type: "string"
+          minLength: 1
+          maxLength: 25
+      required:
+        - platform
+        - osName
+        - architecture
+
+  artifacts:
+    type: "array"
+    items:
+      type: "object"
+      properties:
+        type:
+          type: "string"
+          enum: ["json", "text"]
+        name:
+          description: |
+            The artifact name can be anything.  But when the name ``text_log_summary`` is used
+            then treeherder uses that as the summary file for the Log Viewer in the UI.
+          type: "string"
+          minLength: 1
+          maxLength: 50
+        blob:
+          type: "string"
+          minLength: 1
+          maxLength: 4096
```

Comment:
For log files, we just hold the reference to it, not the log itself.  But for the new error summary files coming in with auto-starring, it is possible for them to be quite large.  Currently, we have no limit on size for an artifact, and perhaps we shouldn't put a limit yet.